src/AdornedRulerPanel.cpp

```diff
@@ -1282,9 +1282,9 @@ AdornedRulerPanel::AdornedRulerPanel(AudacityProject* project,
 
    mOuter = GetClientRect();
 
-   mRuler.SetUpdater( std::make_unique<LinearUpdater>( mRuler, mViewInfo ), mLeftOffset );
+   mRuler.SetUpdater( std::make_unique<LinearUpdater>( mViewInfo ), mLeftOffset );
    mRuler.SetLabelEdges( false );
-   mRuler.SetFormat( Ruler::TimeFormat );
+   mRuler.SetFormat( TimeFormat );
```

Comment:
I confirm that at the second commit, you do not introduce dependency cycles among sourc code files.  👍